lib/iris/_data_manager.py

```diff
@@ -63,6 +68,9 @@ def __init__(self, data, realised_dtype=None):
         self._realised_dtype = None
         self._realised_dtype_setter(realised_dtype)
 
+        # Must be set after the realised dtype.
+        self.fill_value = fill_value
```

Comment:
@djkirkham I just wanted to bank these changes as-is ... my follow on PR will address all of this and other similar issues. But I'd prefer to not do that here in this PR, if that's okay with you? Otherwise, it gets bit noisy.